core/domain/blog_statistics_services.py

```diff
@@ -0,0 +1,1094 @@
+        stats_domain_obj)
+    return stats_domain_obj
+
+
+def get_author_blog_post_reads_stats_by_id(
+    author_id: str
+) -> blog_statistics_domain.AuthorBlogPostReadsAggregatedStats:
+    """Retrieves the AuthorBlogPostReadsAggregatedStats domain instance.
+
+    Args:
+        author_id: str. User ID of the author.
+
+    Returns:
+        AuthorBlogPostReadsAggregatedStats. The author blog posts read stats
+        domain object.
+    """
+    stats_model = blog_stats_models.AuthorBlogPostReadsAggregatedStatsModel.get(
+        author_id, strict=True)
+    stats_domain_obj = (
+        blog_statistics_domain.AuthorBlogPostReadsAggregatedStats(
+            author_id,
+            stats_model.reads_by_hour,
+            stats_model.reads_by_date,
+            stats_model.reads_by_month,
+            stats_model.created_on
+        )
+    )
+    add_missing_stat_keys_with_default_values_in_reads_stats(
+        stats_domain_obj)
+    return stats_domain_obj
+
+
+def get_author_blog_posts_reading_time_stats_by_id(
+    author_id: str
+) -> blog_statistics_domain.AuthorBlogPostsReadingTime:
+    """Retrieves the AuthorBlogPostsReadingTime domain instance.
+
+    Args:
+        author_id: str. User ID of the author.
+
+    Returns:
+        AuthorBlogPostsReadingTime. The author blog post reading time stats
+        domain object.
+    """
+    reading_time_stats = (
+        blog_stats_models.AuthorBlogPostAggregatedReadingTimeModel.get(
+            author_id
+        )
+    )
+    return blog_statistics_domain.AuthorBlogPostsReadingTime(
+        author_id,
+        reading_time_stats.zero_to_one_min,
+        reading_time_stats.one_to_two_min,
+        reading_time_stats.two_to_three_min,
+        reading_time_stats.three_to_four_min,
+        reading_time_stats.four_to_five_min,
+        reading_time_stats.five_to_six_min,
+        reading_time_stats.six_to_seven_min,
+        reading_time_stats.seven_to_eight_min,
+        reading_time_stats.eight_to_nine_min,
+        reading_time_stats.nine_to_ten_min,
+        reading_time_stats.more_than_ten_min,
+    )
+
+
+def save_blog_post_views_stats_model(
+    views_stats: blog_statistics_domain.BlogPostViewsAggregatedStats
+) -> None:
+    """Updates the BlogPostViewsAggregatedStatsModel datastore instance with the
+    passed BlogPostViewsAggregatedStats domain object.
+
+    Args:
+        views_stats: BlogPostViewsAggregatedStats. The blog post views
+            aggregated stats domain object.
+
+    Raises:
+        Exception. No blog post views stats model exists for the given
+            blog_post_id.
+    """
+    blog_post_views_aggregated_stats = (
+        blog_stats_models.BlogPostViewsAggregatedStatsModel.get(
+            views_stats.blog_post_id, strict=False
+        )
+    )
+    if blog_post_views_aggregated_stats is None:
+        raise Exception(
+            'No blog post views stats model exists for the given blog_post_id.'
+        )
```

Comment:
```suggestion
        raise Exception(
            'No blog post views stats model exists for the given blog_post_id: %s' %
            views_stats.blog_post_id
        )
```

So that when an exception occurs, it would be clear for which id model does not exist.

Reply:
Done